doc/api/stream.md

```diff
@@ -1852,13 +1852,11 @@ or write buffered data before a stream ends.
 
 #### Errors While Writing
 
-It is recommended that errors occurring during the processing of the
-`writable._write()` and `writable._writev()` methods are reported by invoking
-the callback and passing the error as the first argument. This will cause an
-`'error'` event to be emitted by the `Writable`. Throwing an `Error` from within
-`writable._write()` can result in unexpected and inconsistent behavior depending
-on how the stream is being used. Using the callback ensures consistent and
-predictable handling of errors.
+Errors occurring during the processing of the [`writable._write()`][],
+[`writable._writev()`][] and [`writable._final()`] methods should be propagated
```

Comment:
I would make this stronger with a `must` rather than `should`
```suggestion
[`writable._writev()`][] and [`writable._final()`] methods must be propagated
```

Reply:
and likewise below...

Reply:
I like that. However, the reason I used `should` is so that stream users don't 100% assume this is the case... since the ecosystem might not conform... a typical example of such cases https://github.com/nodejs/node/issues/29699.

Given that you still think we should change I'm all for it.

Reply:
No you are right. No question about it.